src/lib/operators/table_scan/sorted_segment_search.hpp

```diff
@@ -303,13 +359,8 @@ class SortedSegmentSearch {
     }
   }
 
-  // Flags to indicate whether a shortcut was taken to skip scanning.
-  bool no_rows_matching{false};
```

Comment:
Lines above: we typically avoid increments as part of other statements (here: `matches[output_idx++]`). As you are working on the file: could you move a line below (or in the loop header, I am fine with that as well). The compiler will optimize it ÷anyways.